packages/bitcore-node/src/providers/chain-state/internal/internal.ts

```diff
@@ -199,6 +201,27 @@ export class InternalStateProvider implements CSP.IChainStateService {
     }
   }
 
+  async getAuthhead(params: CSP.StreamTransactionParams) {
+    let { chain, network, txId } = params;
+    if (typeof txId !== 'string') {
+      throw 'Missing required param';
+    }
+    const found = (await CoinModel.resolveAuthhead(txId, chain, network))[0];
+    if (found) {
+      const transformedCoins = found.identityOutputs.map<CoinJSON>(output =>
+        CoinModel._apiTransform(output, { object: true })
+      );
+      return {
+        chain: found.chain,
+        network: found.network,
+        authbase: found.authbase,
+        identityOutputs: transformedCoins
+      };
+    } else {
+      return undefined;
```

Comment:
shouldn't there always be an authHead?

Reply:
Yes, but the provided `txId` might not exist.